### PR TITLE
enhancement(prometheus_exporter sink): Add internal events

### LINKED ISSUE #4948: Add internal events to prometheus_exporter sink
This sink does not appear to be instrumented yet.

Requested by user in discord: https://discord.com/channels/742820443487993987/746070591097798688/775755348430684160

### PR DESCRIPTION
**Pull request by bruceg**
Closes #4948 

Signed-off-by: Bruce Guenter <bruce.guenter@datadoghq.com>

### CODE REVIEW

File: docs/reference/components/sinks/prometheus_exporter.cue [hunk 1]
```diff
@@ -371,4 +371,10 @@ components: sinks: prometheus_exporter: {
 				"""
 		}
 	}
+
+	telemetry: metrics: {
+		processed_bytes_total:  components.sources.internal_metrics.output.metrics.processed_bytes_total
+		processed_events_total: components.sources.internal_metrics.output.metrics.processed_events_total
+		request_errors_total:   components.sources.internal_metrics.output.metrics.request_errors_total
```
Comment (binarylogic) [inline docs/reference/components/sinks/prometheus_exporter.cue:378]: Should we also have one for successes?

**Comment by binarylogic** [comment]
Thanks, @bruceg. To clarify a little bit, I wanted you to do one sink to set precedence. I assumed the first sink would be much more involved since we would need to change sink-level constructs to comprehensively instrument sinks (ex: request libraries). This way all sinks that use the constructs get free telemetry. Do you think that is outside the scope of this issue? Should we open other issues for that? I thought it would be easier to do this in the context of an HTTP-based sink.

**Comment by jszwedko** [comment]
I'll note that we do plan to add `events_in_total` at the topology layer (https://github.com/timberio/vector/issues/6367).

This sink is a bit odd in that it fires internal events when scraped which doesn't really match push-based sink's idea of `processed_events_total`. I think a user may be confused that their `prometheus_exporter` component has less `processed_events_total` than the sum of its inputs. I think we might prefer to:

* Emit `processed_events_total` when incoming metrics events have simply been handled by the sink and but into internal state
* Drop `processed_bytes_total` since I don't think it is meaningful here
* Emit  `exported_metrics_total` when an incoming HTTP request fetches metrics
* Emit `exported_bytes_total` (or maybe `net_out_bytes_total`) when an incoming HTTP request fetches metrics

We could also consider just not emitting `processed_events_total` since `events_in_total` should be added shortly.

**Comment by bruceg** [comment]
> To clarify a little bit, I wanted you to do one sink to set precedence.

Ah, ok, I was missing this context. However, for this task the `prometheus_exporter` is probably the most awkward choice, both because it is works the reverse of all the other sink types and because metrics potentially get coalesced when buffered. Almost any other logs would be better to set precedent, I think.

What you describe, putting it into the request libraries etc, makes sense for most of the sinks, given how many systems have converged on HTTP transports. I think it would be better to make a separate issue to set the precedent and instrument the core traits, and then circle back to `prometheus_exporter` once the pattern is set.

**Comment by ktff** [comment]
> I'll note that we do plan to add events_in_total at the topology layer (#6367).

`events_in_total` and `events_out_total` have been added some time ago #6433, that of course includes this sink.

**Comment by bruceg** [comment]
Given the above feedback and subsequent internal observability work, this isn't the path we want to take any more, so I will close it out.

**Pull request closed without merging**